test/aria/widgets/wai/datePicker/readDate/Base.js

```diff
@@ -0,0 +1,167 @@
+            // --------------------------------------------------- destructuring
+
+            var document = Aria.$window.document;
+
+            var confirmDateDelay = this.confirmDateDelay;
+
+            var data = this._getData();
+            var elements = data.elements;
+
+            // ------------------------------------------------------ processing
+
+            this._executeStepsAndWriteHistory(callback, function (api) {
+                // ----------------------------------------------- destructuring
+
+                var step = api.addStep;
+                var says = api.addToHistory;
+                var delay = api.addDelay;
+
+                // --------------------------------------------- local functions
+
+                function selectStartPoint() {
+                    step(['click', document.getElementById(elements.before.id)]);
+                }
+
+                function pressKey(key) {
+                    step(['type', null, key]);
+                }
+
+                function pressSpecialKey(key) {
+                    pressKey('[' + key + ']');
+                }
+
+                function down() {pressSpecialKey('down');}
+                function enter() {pressSpecialKey('enter');}
+                function right() {pressSpecialKey('right');}
+
+                // -------------------------------------------------- processing
+
+                api.defaultDelay = confirmDateDelay * 2;
+
+                // -------------------------------------------------------------
+                // Navigating to the DatePicker's calendar button
+
+                selectStartPoint();
+                // filtered
+                // says('focus me');
+
+                down();
+                says('calendar one label');
+                down();
+                says('Edit 6/9/16');
+                down();
+                says('calendar one button', 'button menu collapsed');
+
+                // -------------------------------------------------------------
+                // Opening calendar and selecting the next date
+
+                enter();
+                says('calendar one');
+                says('6 September 2016');
+
+                right();
+                says('7 September 2016');
+
+                // -------------------------------------------------------------
+                // Typing enter to validate
+
+                enter();
+                delay(7000);
+                says('calendar one label Edit');
+                says('7/9/16', 'Type in text.'); // input content is read
+                says('Wednesday 7 September 2016'); // date is read
+
+                // -------------------------------------------------------------
+                // Inputting date "by hand"
+
+                pressKey('[backspace]5');
+                says('6'); // removing the last letter
+                // I don't know why this letter gets read but doesn't appear in the speech history
```

Comment:
This is something general: Jaws reads all keys pressed by the user but does not include them in the speech history.